src/controllers/qasm_controller.hpp

```diff
@@ -730,6 +726,28 @@ size_t QasmController::required_memory_mb(
   }
 }
 
+Transpile::Fusion QasmController::transpile_fusion(Method method,
+                                                   const Circuit &circ,
```

Comment:
Originally when I added this I was checking the number of qubits in the circuit, but ended up leaving that to the fusion pass itself. In principle you might want to decide the fusion pass parameters based on the circuit, but this can be removed for now an added back later if needed.